src/org/omegat/gui/scripting/ScriptingWindow.java

```diff
@@ -567,17 +572,23 @@ protected void done() {
     }
 
     public void executeScript(String scriptString, ScriptItem scriptItem) {
-        executeScript(scriptString, scriptItem, Collections.emptyMap());
+    	cancelScriptQueue();
+        ScriptWorker worker = createScriptWorker(scriptString, scriptItem, Collections.emptyMap());
+        currentScriptWorker = worker;
+        worker.execute();
     }
 
     public void executeScriptFile(ScriptItem scriptItem) {
         executeScriptFile(scriptItem, Collections.emptyMap());
     }
 
     public void executeScriptFile(ScriptItem scriptItem, Map<String, Object> additionalBindings) {
+    	cancelScriptQueue();
         try {
             String scriptString = scriptItem.getText();
-            executeScript(scriptString, scriptItem, additionalBindings);
+            ScriptWorker worker = createScriptWorker(scriptString, scriptItem, additionalBindings);
+            currentScriptWorker = worker;
+            worker.execute();
```

Comment:
Why not continue to call `executeScript(String, ScriptItem)` here instead of duplicating its contents?